modules/barcode/include/opencv2/barcode.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 OpenCV Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef __OPENCV_BARCODE_HPP__
+#define __OPENCV_BARCODE_HPP__
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+namespace barcode {
+class CV_EXPORTS_W BarcodeDetector
+{
+public:
+    CV_WRAP BarcodeDetector();
+
+    ~BarcodeDetector();
+
+    /** @brief Detects Barcode in image and returns the rectangle(s) containing the code.
+
+    @param img grayscale or color (BGR) image containing (or not) Barcode.
+    @param rects Output vector of rotated rectangle(s) containing the barcode.
+     */
+    CV_WRAP bool detect(InputArray img, CV_OUT std::vector <RotatedRect> &rects) const;
```

Comment:
> RotatedRect

RotatedRect is not a best option to handle perspective transformations.

Consider switching on quadrangle result:
- 4 points through `std::vector<cv::Point>` (or `std::vector<cv::Point2f>`).
    Note: `std::array` is not supported by binding generator.
- fix order of these points (in meaning of bl/tl/tr/br) to avoid unexpected 90/180 degree rotation

These 4 points can be used with `getPerspectiveTransform()` + `warpPerspective()` to restore "plain" barcode for decoding.

Reply:
We had replaced the `std::vector <RotatedRect>` by `vector<vector<Point2f>>` with Specific order of points (bl,tl,tr,br) in the commit [ad603e0](https://github.com/opencv/opencv_contrib/pull/2757/commits/ad603e0fe5c557380e8a36d1ae2f24f365150f88)